frappe/oauth.py

```diff
@@ -240,9 +240,8 @@ def invalidate_authorization_code(self, client_id, code, request, *args, **kwarg
 	def validate_bearer_token(self, token, scopes, request):
 		# Remember to check expiration and scope membership
 		otoken = frappe.get_doc("OAuth Bearer Token", token)
-		token_expiration_local = otoken.expiration_time.replace(
-			tzinfo=pytz.timezone(get_system_timezone())
-		)
+		tzinfo = pytz.timezone(get_system_timezone())
+		token_expiration_local = tzinfo.localize(otoken.expiration_time)
```

Comment:
All the timezone dance in this function seems completely unnecessary, we can directly compare the datetime fields. 

Am I missing anything here?

Reply:
Sorry, your message is not clear. 

Are you saying that, we may simply compare datetimes directly without having any timezone?